scripts/import_pressbooks.py

```diff
@@ -0,0 +1,149 @@
+            'identifiers': '',
+            'code': '',
+        },
+    ).json()
+}
+
+
+def convert_pressbooks_to_ol(data):
+    book = {"source_records": ['pressbooks:%s' % data['url']]}
+    if data.get('isbn'):
+        book['isbn_13'] = [
+            isbn.split(' ')[0].replace('-', '') for isbn in data['isbn'].split('; ')
+        ]
+    if data.get('name'):
+        book['title'] = data['name']
+    if data.get('languageCode'):
+        book['languages'] = [langs[data['languageCode'].split('-', 1)[0]]]
+    if data.get('author'):
+        book['authors'] = [{"name": a} for a in data.get('author')]
+    if data.get('image'):
+        book['cover'] = data['image']
+    description = (
+        (data.get('description') or '')
+        + '\n\n'
+        + (data.get('disambiguatingDescription') or '')
+    ).strip()
+    if description:
+        book['description'] = description
+    if data.get('alternateName'):
+        book['other_titles'] = [data['alternateName']]
+    if data.get('alternativeHeadline'):
+        book['edition_name'] = data['alternativeHeadline']
+    book['publish_date'] = (
+        data.get('datePublished')
+        or data.get('copyrightYear')
+        or datetime.datetime.fromtimestamp(data.get('lastUpdated')).date().isoformat()
+    )
+    assert book['publish_date'], data
+
+    subjects = (data.get('about') or []) + (data.get('keywords') or '').split(', ')
+    if subjects:
+        book['subjects'] = [
+            s.strip().capitalize() for s in subjects if s  # Sometimes they're null?
+        ]
+
+    book['publishers'] = [p for p in (data.get('networkName'), "Pressbooks") if p]
+
+    book['providers'] = [
+        {
+            'provider': 'pressbooks',
+            'url': data['url'],
+        }
+    ]
+    book['physical_format'] = 'Ebook'
+
+    copyright_line = ' '.join(
+        [
+            data.get('copyrightYear') or '',
+            data.get('copyrightHolderName') or '',
+        ]
+    ).strip()
+    if copyright_line:
+        book['copyright_date'] = copyright_line
+
+    if data.get('wordCount'):
+        book['word_count'] = data['wordCount']
+
+    contributors_map = {
+        'translator': 'Translator',
+        'editor': 'Editor',
+        'illustrator': 'Illustrator',
+        'reviewedBy': 'Reviewer',
+        'contributor': 'Contributor',
+    }
+
+    contributors = [
+        [
+            {"name": person, "role": ol_role}
+            for person in (data.get(pressbooks_field) or [])
+        ]
+        for pressbooks_field, ol_role in contributors_map.items()
+    ]
+    contributors = [contributor for lst in contributors if lst for contributor in lst]
+
+    if contributors:
+        book['contributors'] = contributors
+
+    return book
+
+
+def main(ol_config: str, filename: str, batch_size=5000, dry_run=False):
+
+    if not dry_run:
+        load_config(ol_config)
+        date = datetime.date.today()
+        batch_name = "%s-%04d%02d" % ('pressbooks', date.year, date.month)
```

Comment:
```suggestion
        batch_name = f"pressbooks-{date:%Y}{date:%m}"
```

Reply:
little tweak; tested and this works as well!
```suggestion
        batch_name = f"pressbooks-{date:%Y%m}"
```